Fix potential u3 gate bug.

nfwvogt opened the pull request:
There seems to be a typo in the U3-gate application, using lambda instead of phi in the phase of the imaginary part of the row 2, column 1 element of the U3 matrix.

smu160 commented:
Hi @nfwvogt,

Thank you for submitting this fix, and apologies for the belated response! I was working on merging in all of the parallel code and a bunch of other updates into main, so we can now proceed with this fix.

Would you be able to apply your bug fix to the current version of main? In addition, would you be able to provide your test case? I would like to add it to the tests since it did a better job of spotting the issue than the single test I have for $U$ at the moment. I'd be glad to add in the test to this PR, and then we can merge it in.

Thank you!

nfwvogt committed:
Ported bugfix

nfwvogt commented:
I have rebased the branch and added a simple unittest for u3 on just one qubit. The previous unittest didn't catch the issue because phi and lambda are equal in that one.

smu160 commented:
@nfwvogt Awesome. Thank you!
